pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -736,8 +737,14 @@ private void setupOutput(ContextImpl contextImpl) throws Exception {
                 pulsarSinkConfig.setProcessingGuarantees(FunctionConfig.ProcessingGuarantees.valueOf(
                         this.instanceConfig.getFunctionDetails().getProcessingGuarantees().name()));
                 pulsarSinkConfig.setTopic(sinkSpec.getTopic());
-                pulsarSinkConfig.setForwardSourceMessageProperty(
+                
+                if (this.instanceConfig.getFunctionDetails().getComponentType()== ComponentType.SOURCE) {
+                  // Forward record level properties for all sources by default
+                  pulsarSinkConfig.setForwardSourceMessageProperty(true);
+                } else if (this.instanceConfig.getFunctionDetails().getComponentType() == ComponentType.SINK) {
```

Comment:
Can this be a simple `else` ? ( without `if`)

Reply:
I cannot because there are 4 different component type enums:

enum ComponentType {
        UNKNOWN = 0;
        FUNCTION = 1;
        SOURCE = 2;
        SINK = 3;
    }

Reply:
I more comprehensive solution would be to add a `ForwardSourceMessageProperty` to the configuration for ALL component types and use a default value of `true` for all type?  Seems like from a consistency perspective having all the components behave exactly the same w.r.t to forward properties but leave an option for the user to override this behavior as need.

Thoughts?

Reply:
@eolivelli I have replaced the code with a different approach. If you have time can you please review the updated code and mark this issue as resolved or suggest what other changes are required to address you concerns? Thanks in advance!